GovUk.Frontend.AspNetCore.Extensions.ConformanceTests/Assert.cs

```diff
@@ -0,0 +1,82 @@
+using AngleSharp.Diffing;
+using AngleSharp.Diffing.Core;
+using AngleSharp.Dom;
+using System.Text;
+
+namespace GovUk.Frontend.AspNetCore.Extensions.ConformanceTests
+{
+    internal static class Assert
+    {
+        internal static void HtmlEqual(
+            string expected,
+            string actual)
+        {
+            var diffs = DiffBuilder.Compare(expected).WithTest(actual).Build()
+                .Where(diff => !ExcludeFalsePositive(diff))
+                .ToArray();
+
+            if (diffs.Length != 0)
+            {
+                var sb = new StringBuilder();
+                sb.AppendLine($"{nameof(Assert)}.{nameof(HtmlEqual)}() failure");
+                sb.AppendLine();
+
+                foreach (var diff in diffs)
+                {
+                    DiffConverter.Append(diff, sb);
+                }
+
+                throw new InvalidOperationException(sb.ToString());
```

Comment:
Throwing an `InvalidOperationException()` here makes me think there is something wrong with the test code itself, maybe the test code is in a state where this class doesn't know how to make an assertion. 

Instead, we could use NUnits `Assert.Fail(sb.ToString());`. This generates a more uniform message in Test explorer and would be the message I would expect to see if my implementation of a component no longer conforms

Reply:
Agreed and updated.